Mention how to immediately remove cookies (#41354)
mdn/content: files/en-us/web/http/guides/cookies/index.md

  > [!NOTE]
  > If your site authenticates users, it should regenerate and resend session cookies, even ones that already exist, whenever a user authenticates. This approach helps prevent [session fixation](https://owasp.org/www-community/attacks/Session_fixation) attacks, where a third-party can reuse a user's session.

To immediately remove a cookie, set the cookie again with the same name, path, and domain (if specified), and set its `Expires` attribute to a date in the past or its `Max-Age` attribute to `0` or negative. This instructs the browser to delete the cookie right away. For example:

```http
Set-Cookie: id=a3fWa; Max-Age=0
```

> 

Author: I am unsure of the scope of CSD. I'm pretty sure it's only limited to subdomains (the spec says it uses "domain-match"), but both your edit here and the comment in the original issue suggest that it's to everything under the same TLD+1.

> 

Reviewer: You are right to be unsure. I believe the comment to be true based on this part of the spec

> When delivered with a response from https://example.com/clear, the following header will cause cookies associated with the origin https://example.com to be cleared, as well as cookies on any origin in the same **registered domain** (e.g. https://www.example.com/ and https://more.subdomains.example.com/).

A registered domain is the top level domain, which matches my text.

@Jxck Can you confirm this point I am adding in https://github.com/mdn/content/pull/41354#discussion_r2400462543 - i.e that clear affects the whole domain irrespective of which subdomain it is served from.

> 

Reviewer 2: It's called a "registrable domain" (see URL standard). I see that Clear-Site-Data says "registered domain", but that's wrong. In any event, in terms of scope it's correct.

https://w3c.github.io/webappsec-clear-site-data/#clear-cookies
https://url.spec.whatwg.org/#host-registrable-domain

  > [!NOTE]
  > If your site authenticates users, it should regenerate and resend session cookies, even ones that already exist, whenever a user authenticates. This approach helps prevent [session fixation](https://owasp.org/www-community/attacks/Session_fixation) attacks, where a third-party can reuse a user's session.

To immediately remove a cookie, set the cookie again with the same name, path, and domain (if specified), and set its `Expires` attribute to a date in the past or its `Max-Age` attribute to `0` or negative. This instructs the browser to delete the cookie right away. For example:

```http
Set-Cookie: id=a3fWa; Max-Age=0
```

> It is also possible to clear all cookies associated with a registrable domain using the {{httpheader("Clear-Site-Data")}} response header.

Author: ```suggestion
You can also clear all cookies associated with a registrable domain using the {{httpheader("Clear-Site-Data")}} response header.
```
